File: ios/RNMockBlocking.m
//
//  RNMockBlocking.m
//  ReactNativeWithKMP
//
//  Created by MacPro2 on 2024/6/27.
//

#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(RNMockBlocking, NSObject)

RCT_EXTERN_METHOD(mockBlocking:
                  (nonnull NSNumber *) duration)

@end

File: ios/RNMockBlocking.swift
//
//  RNMockBlocking.swift
//  ReactNativeWithKMP
//
//  Created by MacPro2 on 2024/6/27.
//

import Foundation

@objc(RNMockBlocking)
class RNMockBlocking: RCTEventEmitter {
  
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  @objc(mockBlocking:)
  func mockBlocking(_ duration: NSNumber) {
    DispatchQueue.main.async {
      sleep(UInt32(duration.int32Value) / 1000)
    }
  }
}
